public/app/core/components/AppChrome/MegaMenu/MegaMenu.tsx

```diff
@@ -46,6 +50,29 @@ export const MegaMenu = memo(
       });
     };
 
+    const isPinned = useMemo(
+      (id?: string) => {
+        if (!id || !pinnedItems?.length) {
+          return false;
+        }
+        return pinnedItems?.includes(id);
+      },
+      [pinnedItems]
+    );
+
+    const onPinItem = (id?: string) => {
+      if (id) {
```

Comment:
Are there any menu items that don't have ids?

Reply:
haven't seen any but the id is marked as optional in the NavLinkItem type.